modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetContext.java

```diff
@@ -247,4 +246,11 @@ public long updateLastStateUpdatedTime() {
         this.lastStateUpdatedTime = System.currentTimeMillis();
         return this.lastStateUpdatedTime;
     }
+
+    public static Boolean isCorrelationIdAvailable(NHttpConnection connection) {
```

Comment:
Can the connection or the context be null at this point?

Reply:
fixed with https://github.com/wso2/wso2-synapse/pull/1397/commits/4ecdd6e044a7436c57bf20df03ecba59bf9d23aa